framework/src/utils/lib/data-vpc-props.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+import { RemovalPolicy } from 'aws-cdk-lib';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { IKey } from 'aws-cdk-lib/aws-kms';
+import { RetentionDays } from 'aws-cdk-lib/aws-logs';
+
+/**
+ * The properties for the DataVpc construct
+ */
+
+export interface DataVpcProps {
+  /**
+   * The CIDR to use to create the subnets in the VPC.
+   */
+  readonly vpcCidr: string;
+  /**
+   * The KMS key for the VPC flow log group
+   * @default - A new KMS key is created
+   */
```

Comment:
Can you add a note to ask the user to add a resource policy to the key to allow its use by the flow log?

https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/encrypt-log-data-kms.html